src/odemis/gui/cont/views.py

```diff
@@ -373,11 +376,19 @@ def _initialize_ovv_im(self, shape):
     def _on_merge_ratio_change(self, ratio):
         self.canvas.history_overlay.set_merge_ratio(ratio)
 
-    def on_stage_pos_change(self, p_pos):
+    def on_stage_pos_change(self, pos):
         """ Store the new position in the overview history when the stage moves,
         update the overview image """
 
-        # History
+        # Wait 0.5 seconds before adding the position to the history to get fewer
+        # updates while moving and avoid a cluttered overview
+        wx.CallAfter(self._timer_pos.Stop)
+        wx.CallAfter(self._timer_pos.Start, milliseconds=500, oneShot=True)
+        self._update_ovv()
```

Comment:
I'm still wondering whether this `update_ovv()` is necessary. Could you either add a comment why it's necessary, or remove it?